tree/checkMirrorImage.js

```diff
@@ -0,0 +1,20 @@
+const BinaryTree = require("./BinaryTree");
+
+let checkMirrorImage = (tree1, tree2) => {
+    let isMirrorImage = true;
+    checkIfMirrorImage = (tree1, tree2)=> {
+        isMirrorImage = isMirrorImage && ((tree1 && tree2) || (!tree1 && !tree2));
+        isMirrorImage = isMirrorImage && (tree1.value === tree2.value);
+
+        if(isMirrorImage) {
+            checkMirrorImage(tree1.left, tree2.right);
+            checkMirrorImage(tree1.right, tree2.left);
+        }
+
+        return;
+    }
+    return isMirrorImage;
+}
+
+module.exports = checkMirrorImage;
+
```

Comment:
This could possibly work
```javascript
let isMirrorImage = (tree1, tree2) => {
    const areBothTreesNull = !tree1 && !tree2;
    if (areBothTreesNull) return true;
    const areBothTreesNonNull = tree1 && tree2;
    if (!areBothTreesNonNull) return false;
    const areRootValuesSame = tree1.value === tree2.value
    if (areRootValuesSame) {
        return checkMirrorImage(tree1.left, tree2.right) && checkMirrorImage(tree1.right, tree2.left);
    } else {
        return false
    }
}
```

Reply:
wont this fail if i find a mismatch at a level and after it returned false, due to recursion it will keep running and if no other mismatch found, will return true.

Reply:
recursion is happening at this step, and this will only run if first 3 checks pass otherwise it should not proceed.
```javascript
return checkMirrorImage(tree1.left, tree2.right) && checkMirrorImage(tree1.right, tree2.left);
``` 
I have not tested it, this is just a rough program but having an outside scoped variable is way worse.